tests/integration/telemetry/api/setup_test.go

```diff
@@ -161,8 +162,16 @@ proxyMetadata:
 // By looking into a context name. Expects "kind-" prefix.
 // That is required by some tests for specific actions on "Kind".
 func IsKindCluster() (bool, error) {
-	config, err := clientcmd.LoadFromFile(clientcmd.RecommendedHomeFile)
+	kubeconfig := os.Getenv("KUBECONFIG")
```

Comment:
Hmm, thinking about this more -- this is looking at the current context pointed at, but that isn't always how the test is configured. For multiclsuter, for instance, ` --istio.test.kube.topology=opology-config.json` is used to point to a set of clusters that are unrelated to the current context.

The most reliable way would be to somehow get this from `(resource.Context).Clusters()`. I don't know how to get from there to the info we need, though.

Reply:
From what I could see, the multicluster test has been executed as part of this gate.
And I could see that it loads the cluster topology as - `/home/prow/go/src/istio.io/istio/prow/config/topology/multicluster.json`. And it looks like it is not affected by this change.

Maybe I could modify the logic of the function, so in case of an error to load the kubeconfig file, it will return `true` as a fallback to be indicated as kind cluster.

Reply:
I am not concerned with the CI we have -- it clearly works, likely because it happens to point to the kind cluster it created. But in theory you can (and I have) run things locally like this in a manner it wouldn't work.

I am 100% fine with not handling every obscure edge case :slightly_smiling_face: just wanted to make it known that there are edge cases

Reply:
You're right.

Unfortunately, as per now, I could not think of any proper solution.
But from the other side, without this PR, CI prow Openshift job will fail during execution. :(

Reply:
Where could I find the example structure  of `--istio.test.kube.topology=opology-config.json` file?
I will try to see how it work with this patch and what could be done with it.